plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+{
+    struct mk_http_header *auth_header = NULL;
+
+    if (ctx->auth_header == NULL) {
+        return SPLUNK_AUTH_UNAUTH;
+    }
+
+    auth_header = mk_http_header_get(MK_HEADER_AUTHORIZATION, request, NULL, 0);
+
+    if (auth_header == NULL) {
+        return SPLUNK_AUTH_MISSING_CRED;
+    }
+
+    if (auth_header && auth_header->val.len > 0) {
+        if (strncmp(ctx->auth_header, auth_header->val.data, strlen(ctx->auth_header)) == 0) {
+            return SPLUNK_AUTH_SUCCESS;
+        }
+        else {
+            return SPLUNK_AUTH_UNAUTHORIZED;
+        }
+    }
+    else {
+        return SPLUNK_AUTH_MISSING_CRED;
+    }
+
+    return SPLUNK_AUTH_SUCCESS;
+}
+
+
+static int process_hec_payload(struct flb_splunk *ctx, struct splunk_conn *conn,
+                               flb_sds_t tag,
+                               struct mk_http_session *session,
+                               struct mk_http_request *request)
+{
+    int i = 0;
+    int ret = 0;
+    int type = -1;
+    struct mk_http_header *header;
+    int extra_size = -1;
+    struct mk_http_header *headers_extra;
+    int gzip_compressed = FLB_FALSE;
+    void *gz_data = NULL;
+    size_t gz_size = -1;
+
+    header = &session->parser.headers[MK_HEADER_CONTENT_TYPE];
+    if (header->key.data == NULL) {
+        send_response(conn, 400, "error: header 'Content-Type' is not set\n");
+        return -1;
+    }
+
+    if (header->val.len == 16 &&
+        strncasecmp(header->val.data, "application/json", 16) == 0) {
+        type = HTTP_CONTENT_JSON;
+    }
+
+    if (header->val.len == 10 &&
+        strncasecmp(header->val.data, "text/plain", 10) == 0) {
+        type = HTTP_CONTENT_TEXT;
+    }
+
+    if (type == -1) {
+        /* Not neccesary to specify content-type for Splunk HEC. */
+        flb_plg_debug(ctx->ins, "Mark as unknown type for ingested payloads");
+        type = HTTP_CONTENT_UNKNOWN;
+    }
+
+    if (request->data.len <= 0) {
+        send_response(conn, 400, "error: no payload found\n");
+        return -1;
+    }
+
+    extra_size = session->parser.headers_extra_count;
+    if (extra_size > 0) {
+        for (i = 0; i < extra_size; i++) {
+            headers_extra = &session->parser.headers_extra[i];
+            if (headers_extra->key.len == 16 &&
+                strncasecmp(headers_extra->key.data, "Content-Encoding", 16) == 0) {
+                if (headers_extra->val.len == 4 &&
+                    strncasecmp(headers_extra->val.data, "gzip", 4) == 0) {
+                    flb_debug("[splunk_prot] body is gzipped");
+                    gzip_compressed = FLB_TRUE;
+                }
+            }
+        }
+    }
+
+    if (gzip_compressed == FLB_TRUE) {
+        ret = flb_gzip_uncompress((void *) request->data.data, request->data.len,
+                                  &gz_data, &gz_size);
+        if (ret == -1) {
+            flb_error("[elasticsearch_bulk_prot] gzip uncompress is failed");
```

Comment:
Use the `flb_plg_*` logging functions.

Reply:
Sure, we should use this style of logger if possible. Fixed in https://github.com/fluent/fluent-bit/pull/7349/commits/0d9937e63b02acddc47244af835485c3dd58abd8.